### PR TITLE
test(source-medium): setup test on sourceNodes

### PR DESCRIPTION
**Pull request by Slashgear**
## Description

- setup tests with jest on `gatsby-source-medium`
- mock HTTP call
- use my own api data as fixture
- ~sadly, I don't think this package is still working now, medium API seems to be different now => look a created node in snapshot~

### CODE REVIEW

**Comment by pieh** [comment]
> * sadly, I don't think this package is still working now, medium API seems to be different now => look a created node in snapshot

Hmmm, I don't really know the status of package ... but if it's not working we definitely should mark it as such. I don't know if there is new API or something that can fix it or if medium decided to not allow use cases like ours.

I don't really know what to do about tests in scenario like this - it appears there is _some_ data (user information) but not really content? Is that correct?

**Comment by pieh** [comment]
Oh, I just checked snapshot more carefully - so we have links to post nodes ... but we don't have post nodes themselves?

File: packages/gatsby-source-medium/src/__tests__/gatsby-node.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const axios = require(`axios`)
+const { sourceNodes } = require(`../gatsby-node`)
+
+const fixture = require(`./__fixtures__/medium.json`)
+const fakeMediumData = `])}while(1);</x>${JSON.stringify(fixture)}`
```
Comment (pieh) [inline packages/gatsby-source-medium/src/__tests__/gatsby-node.js:5]: uuu wat? :P

Is this how actual responses from medium look like? :)

**Comment by pieh** [comment]
Was doing some checking with https://github.com/gatsbyjs/gatsby/tree/master/examples/using-medium and this seems to "work" (well at least I can run queries and get some details about posts there - still not full content). Readme of package mention issues about content (only preview available) in opening text https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-medium/README.md so it would appear that this part is correct? It feels to me like we can go ahead with tests, but also try to figure out why there is so little nodes created? Maybe we should use configuration from `using-medium` to generate our fixture? Worth trying to see how things change?

File: packages/gatsby-source-medium/src/__tests__/gatsby-node.js (hunk 1 above)
Comment (Slashgear) [reply packages/gatsby-source-medium/src/__tests__/gatsby-node.js:5]: Yep Medium is prefixing their API with `])}while(1);</x>` and I don't know why x)

**Comment by Slashgear** [comment]
I was only testing the first createNode calls 😅
Fix it by snapshoting all calls.

**Pull request merged**